telemetry/jetbrains/src/main/kotlin/software/aws/toolkits/telemetry/generator/TelemetryGenerator.kt

```diff
@@ -128,29 +162,27 @@ object TelemetryGenerator {
             .addStatement("unit(%M.${(metric.unit ?: MetricUnit.NONE).name})", metricUnit)
             .addStatement("value(value)")
         metric.metadata?.forEach {
-            if (it.required == false) {
-                functionBuilder.beginControlFlow("if(%L != null) {", it.type.toArgumentFormat())
-            }
-            functionBuilder.addStatement("metadata(%S, %L.toString())", it.type.toArgumentFormat(), it.type.toArgumentFormat())
-            if (it.required == false) {
-                functionBuilder.endControlFlow()
+            if (boolResultField && it.type == RESULT) {
```

Comment:
Would like some suggestions for not having this bool result field, generating function statements is really tied to the function and it's hard to invert where it's done without it looking worse than this.